src/praisonai-agents/praisonaiagents/agent/retry_utils.py

```
@@ -0,0 +1,90 @@
+    
+    Example:
+        >>> # Attempt 0: ~5s, Attempt 1: ~10s, Attempt 2: ~20s
+        >>> delay = jittered_backoff(1, base_delay=5.0, max_delay=120.0, jitter_ratio=0.5)
+    """
+    # Exponential backoff: base * 2^attempt
+    delay = min(base_delay * (2 ** max(0, attempt)), max_delay)
+    
+    # Apply jitter: delay ± (jitter_ratio * delay)
+    if jitter_ratio > 0:
+        jitter_range = delay * jitter_ratio
+        jitter = random.uniform(-jitter_range, jitter_range)
```

Ruff (0.15.15)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)